## Q1: What are the Probability Mass Function (PMF) and Probability Density Function (PDF)? Explain with an example.

PMF:
1. The probability for each possible value x must be between 0 and 1: 0 ≤ P(X = x) ≤ 1.
2. The sum of probabilities for all possible values must equal 1: Σ P(X = x) = 1, where the sum is taken over all possible values of X.

PDF:
1. The value of the PDF for any specific value of x is non-negative: f(x) ≥ 0.
2. The total area under the PDF curve over the entire range of possible values is equal to 1: ∫ f(x) dx = 1, where the integral is taken over the entire range of X.
3. used to describe the shape and characteristics of continuous probability distributions
4. It allows us to calculate probabilities of events, determine the likelihood of observing a specific range of values

In [2]:
# example of pmf
from scipy.stats import binom

#Define the parameters of binomial distribution
n= 10 #no. of trails
p= 0.5 #probability of success

#caculate pmf a specific point
k=5 # no. od successes

pmf_binom=binom.pmf(k,n,p)
print('pmf value at k=',k,'is',pmf_binom)

pmf value at k= 5 is 0.24609375000000003


In [3]:
# example of pdf
from scipy.stats import norm

#define parameter of normal distribution
mu=2
std=2.5

#calculate pdf at specific point
x=1.5

pdf_norm=norm.pdf(x,mu,std)
print('pdf value at x=',x,'is',pdf_norm)

pdf value at x= 1.5 is 0.15641707759018236


## Q2: What is Cumulative Density Function (CDF)? Explain with an example. Why CDF is used?

The CDF function F(x) gives the probability that the random variable X is less than or equal to x.
1. The CDF allows us to calculate probabilities associated with random variables. 
2. The CDF enables us to find percentiles and quantiles of a distribution.
3.  CDF plays a role in hypothesis testing by allowing us to calculate p-values

In [4]:
# example of cdf
# Let X representing the height of individuals in a population
# We calculate the probability of X being less than or equal to 170 cm 

import scipy.stats as stats

# Define the random variable distribution
rv = stats.norm(loc=160, scale=10)  # Normal distribution with mean 160 and standard deviation 10

# Define the value for which we want to calculate the CDF
x = 170

# Calculate the CDF
cdf = rv.cdf(x)

# Print the result
print("CDF at x =", x, ":", cdf)

CDF at x = 170 : 0.8413447460685429


This means that the CDF at 170 cm is 0.84, indicating that there is an 84% chance of randomly selecting an individual with a height less than or equal to 170 cm.

## Q3: What are some examples of situations where the normal distribution might be used as a model? Explain how the parameters of the normal distribution relate to the shape of the distribution.

The normal distribution, also known as the Gaussian distribution or bell curve, is widely used as a model in various fields
such as Height and Weight, Test scores, Measurements error, Financial markets, etc.

The parameters of the normal distribution determine its shape.
1.  The mean determines the location of the peak of the distribution.
2.  The standard deviation controls the width of the distribution.
3. A smaller standard deviation resulting in a narrower distribution. Conversely, a larger standard deviation leads to a wider distribution with more spread-out data points.
4. By manipulating the mean and standard deviation, the shape of the normal distribution can be adjusted.
5. Shifting the mean to the left or right changes the location of the peak, while changing the standard deviation alters the spread of the distribution.


## Q4: Explain the importance of Normal Distribution. Give a few real-life examples of Normal Distribution.

Importance:
1. Central Limit Theorem: The normal distribution is central to the Central Limit Theorem (CLT), which states that the average of a large number of i.i.d. random variables tends to follow a normal distribution, regardless of the underlying distribution.
2. Many natural processes and human characteristics tend to exhibit a bell-shaped distribution, which can be well described by the normal distribution. Examples include heights and weights of individuals, IQ scores, errors in measurements, and stock market returns.
3. Many statistical techniques and tests are based on the assumption of normality.

Examples:
1. IQ Scores: Intelligence quotient (IQ) scores are often distributed approximately normally, with a mean of 100 and a standard deviation of 15.

2. Heights of Adults: In a population, the heights of adult males or females tend to follow a roughly normal distribution.

3. Test Scores: In educational assessments, test scores often approximate a normal distribution.

4. Errors in Measurements: Measurement errors in various fields, such as scientific experiments, industrial processes, or quality control, are often assumed to be normally distributed.

5. Financial Returns: Price changes or returns in financial markets, such as stock market returns, are frequently modeled using the normal distribution or variations of it.

## Q5: What is Bernaulli Distribution? Give an Example. What is the difference between Bernoulli Distribution and Binomial Distribution?

Bernoulli Distribution:
1. models a random experiment with two possible outcomes: success (usually denoted as 1) or failure (usually denoted as 0).
2. PMF is given by:

   P(X = k) = p^k * (1 - p)^(1 - k)
   where X is the random variable, k is the outcome (0 or 1), and p is the probability of success.
   Example: Tossing a fair coin. Assume P(H) = 0.5 and P(T) = 0.5
   
Difference:
1. Bernoulli distribution models a single trial with two possible outcomes: success or failure.
2. Binomial distribution models the number of successes in a fixed number of independent Bernoulli trials.
    



## Q6. Consider a dataset with a mean of 50 and a standard deviation of 10. If we assume that the dataset is normally distributed, what is the probability that a randomly selected observation will be greater than 60? Use the appropriate formula and show your calculations.

Given dataset with a mean of 50 and a standard deviation of 10.
Let's calculate the z-score:
z = (60 - 50) / 10 = 1
Using a standard normal distribution table, we find that the probability corresponding to a z-score of 1 is approximately 0.8413.
therefore, the probability that a randomly selected observation from the dataset will be greater than 60 is approximately (1-0.8413 =0.1587) or 15.87%.

In [3]:
import scipy.stats as stats

mean = 50
std_dev = 10
value = 60

# Calculate the z-score
z = (value - mean) / std_dev

# Calculate the probability using the survival function (1 - cumulative distribution function)
probability = 1 - stats.norm.cdf(z)

print("Probability:", probability)

Probability: 0.15865525393145707


## Q7: Explain uniform Distribution with an example.

1. The uniform distribution is a continuous probability distribution where all outcomes within a given range are equally likely.
2. The probability density function (PDF) of the uniform distribution is constant within the range and zero outside the range.
3. The probability density function (PDF) of the uniform distribution is given by:
   f(x) = 1 / (b - a)
   where 'a' and 'b' are the lower and upper bounds of the range, respectively.
4. Graphically, the PDF  appears as a constant horizontal line within the defined range [a, b] and zero outside this range.

Example:
let's consider a uniform distribution over the range [0, 10]. In this case, the PDF is:

f(x) = 1 / (10 - 0) = 1/10

This means that any value within the range [0, 10] has a probability density of 1/10.

## Q8: What is the z score? State the importance of the z score.

1. The z-score, also known as the standard score, represents the number of standard deviations an observation or data point is away from the mean of a distribution.
2. The formula to calculate the z-score is:
z = (x - μ) / σ
where z is the z-score, x is the value, μ is the mean of the distribution, and σ is the standard deviation of the distribution.

Importance:
1. Standardization: The z-score use to standardizes data.
2. Relative Position: A positive z-score indicates that the value is above the mean, while a negative z-score indicates that the value is below the mean.
3. Probability Calculation: The z-score is used to calculate probabilities associated with specific values in a distribution.
4. Standard Normal Distribution: The z-score is specifically used in reference to the standard normal distribution, which has a mean of 0 and a standard deviation of 1

## Q9: What is Central Limit Theorem? State the significance of the Central Limit Theorem.

The Central Limit Theorem (CLT) is a fundamental theorem in statistics that states that the distribution of the sample mean from a population with any distribution approaches a normal distribution as the sample size increases, regardless of the shape of the population distribution.

1. The CLT allows us to make inferences about the population mean based on the distribution of sample means
2. as the sample size increases, the sampling distribution of the mean becomes approximately normally distributed
3. even if the population is not normally distributed, as long as the sample size is sufficiently large, the distribution of the sample mean will still approximate a normal distribution.


## Q10: State the assumptions of the Central Limit Theorem.

1. The samples used in the analysis should be selected randomly from the population of interest. 
2.  The observations within each sample should be independent of each other. 
3. The sample size should be sufficiently large.

